clang/include/clang/Tooling/DependencyScanning/ScanAndUpdateArgs.h

```diff
@@ -40,6 +40,9 @@ void configureInvocationForCaching(CompilerInvocation &CI, CASOptions CASOpts,
                                    std::string RootID, std::string WorkingDir,
                                    bool ProduceIncludeTree);
 
+/// Resets some options that introduce dependencies unnecessarily.
+void removeUnnecessaryDependencies(CompilerInvocation &CI, bool ForModuleBuild);
```

Comment:
This should probably live in ModuleDepCollector.h or something similar; it's not specific to caching and we could potentially upstream this part of the change if we found annother reason to use this code.

Reply:
Yeah, good point, I'll do a NFC modification to add it, otherwise we'll have trouble with merge conflicts.

Reply:
See updated commit https://github.com/apple/llvm-project/pull/8564/commits/3b886bc00592494407de5de202a3c7dfadfcb960 (the other commit is an upstream commit that I cherry-picked to the PR, it's not in `next` yet).